<?xml version="1.0"?>
<launch>
  <!-- Object Detection 노드 -->
  <node
    name="object_detection_node"
    pkg="detect_ws"
    type="object_detection_node"
    output="screen"
  >
    <!-- 토픽 파라미터 -->
    <param name="lidar_topic"  value="/livox/lidar"/>
    <param name="camera_topic" value="/camera/image_raw"/>
    <param name="yolo_topic"   value="/yolov8_pub"/>
    <param name="frame_name"   value="livox_frame"/>

    <!-- LiDAR 필터링 범위 -->
    <param name="xMinRange" value="0"/>
    <param name="xMaxRange" value="25"/>
    <param name="yMinRange" value="-6"/>
    <param name="yMaxRange" value="6"/>
    <param name="zMinRange" value="-0.5"/>
    <param name="zMaxRange" value="1.0"/>

    <!-- 클러스터링 파라미터 -->
    <param name="cluster_tolerance" value="1.0"/>
    <param name="cluster_min"       value="4"/>
    <param name="cluster_max"       value="100"/>
  </node>

  <!-- RViz (선택) -->
  <arg name="open_rviz" default="true"/>
  <group if="$(arg open_rviz)">
    <node
      pkg="rviz"
      type="rviz"
      name="rviz"
      required="true"
      output="screen"
      args="-d $(find detect_ws)/rviz/view.rviz"
    />
  </group>
</launch>

